src/Search/LexiconfreeTimesyncBeamSearch/LexiconfreeTimesyncBeamSearch.cc

```diff
@@ -0,0 +1,499 @@
+                    base.trace,
+                    extension.pron,
+                    extension.timeframe + 1,
+                    {extension.score, 0},
+                    {}));
+            break;
+        case Nn::LabelScorer::LABEL_LOOP:
+        case Nn::LabelScorer::BLANK_LOOP:
+            // Copy base trace and update it
+            trace                 = Core::ref(new LatticeTrace(*base.trace));
+            trace->sibling        = {};
+            trace->score.acoustic = extension.score;
+            trace->time           = extension.timeframe + 1;
+            break;
+    }
+}
+
+std::string LexiconfreeTimesyncBeamSearch::LabelHypothesis::toString() const {
+    std::stringstream ss;
+    ss << "Score: " << score << ", traceback: ";
+
+    auto traceback = trace->performTraceback();
+
+    for (auto& item : *traceback) {
+        if (item.pronunciation and item.pronunciation->lemma()) {
+            ss << item.pronunciation->lemma()->symbol() << " ";
+        }
+    }
+    return ss.str();
+}
+
+/*
+ * =====================================
+ * === LexiconfreeTimesyncBeamSearch ===
+ * =====================================
+ */
+
+const Core::ParameterInt LexiconfreeTimesyncBeamSearch::paramMaxBeamSize(
+        "max-beam-size",
+        "Maximum number of elements in the search beam.",
+        1, 1);
+
+const Core::ParameterFloat LexiconfreeTimesyncBeamSearch::paramScoreThreshold(
+        "score-threshold",
+        "Prune any hypotheses with a score that is at least this much worse than the best hypothesis. If not set, no score pruning will be done.",
+        Core::Type<Score>::max, 0);
+
+const Core::ParameterInt LexiconfreeTimesyncBeamSearch::paramBlankLabelIndex(
+        "blank-label-index",
+        "Index of the blank label in the lexicon. Can also be inferred from lexicon if it has a lemma with `special='blank'`. If not set, the search will not use blank.",
+        Core::Type<int>::max);
+
+const Core::ParameterBool LexiconfreeTimesyncBeamSearch::paramCollapseRepeatedLabels(
+        "collapse-repeated-labels",
+        "Collapse repeated emission of the same label into one output. If false, every emission is treated like a new output.",
+        false);
+
+const Core::ParameterBool LexiconfreeTimesyncBeamSearch::paramLogStepwiseStatistics(
+        "log-stepwise-statistics",
+        "Log statistics about the beam at every search step.",
+        false);
+
+LexiconfreeTimesyncBeamSearch::LexiconfreeTimesyncBeamSearch(Core::Configuration const& config)
+        : Core::Component(config),
+          SearchAlgorithmV2(config),
+          maxBeamSize_(paramMaxBeamSize(config)),
+          scoreThreshold_(paramScoreThreshold(config)),
+          blankLabelIndex_(paramBlankLabelIndex(config)),
+          collapseRepeatedLabels_(paramCollapseRepeatedLabels(config)),
+          logStepwiseStatistics_(paramLogStepwiseStatistics(config)),
+          debugChannel_(config, "debug", Core::Channel::standard),
```

Comment:
```suggestion
          debugChannel_(config, "debug"),
```
Otherwise one would have to explicitly suppress the debug output.